JiveAuthenticatingHTTPProtocolDemo/JiveAuthenticatingHTTPProtocolDemo/ViewController.m

```diff
@@ -9,142 +9,269 @@
+}
+
+-(void)loadTestPageThatRequiresAuth
+{
+    NSURL* url = [NSURL URLWithString: @"https://httpbin.org/basic-auth/foo/bar"];
+    
+    NSURLRequest* request = [[NSURLRequest alloc] initWithURL: url];
+    [self.webView loadRequest: request];
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    [self setupWebViewAuthHook];
+    [self loadTestPageThatRequiresAuth];
 }
 
 #pragma mark - JAHPAuthenticatingHTTPProtocolDelegate
 
-- (BOOL)authenticatingHTTPProtocol:(JAHPAuthenticatingHTTPProtocol *)authenticatingHTTPProtocol canAuthenticateAgainstProtectionSpace:(NSURLProtectionSpace *)protectionSpace {
-    BOOL canAuthenticate = [protectionSpace.authenticationMethod isEqualToString:NSURLAuthenticationMethodHTTPBasic];
+- (BOOL)authenticatingHTTPProtocol:(JAHPAuthenticatingHTTPProtocol *)authenticatingHTTPProtocol
+canAuthenticateAgainstProtectionSpace:(NSURLProtectionSpace *)protectionSpace
+{
+    NSArray* interceptedAuthMethods =
+    @[
+      NSURLAuthenticationMethodHTTPBasic
+      , NSURLAuthenticationMethodNTLM
+    ];
+    
+    NSSet* interceptedAuthMethodsSet = [NSSet setWithArray: interceptedAuthMethods];
+    
+    BOOL canAuthenticate =
+        [interceptedAuthMethodsSet containsObject: protectionSpace.authenticationMethod];
+    
     return canAuthenticate;
 }
 
 
 #define USE_PROTOCOL_FOR_CANCELLATION 0
 #if USE_PROTOCOL_FOR_CANCELLATION
-- (JAHPDidCancelAuthenticationChallengeHandler)authenticatingHTTPProtocol:(JAHPAuthenticatingHTTPProtocol *)authenticatingHTTPProtocol didReceiveAuthenticationChallenge:(NSURLAuthenticationChallenge *)challenge {
+- (JAHPDidCancelAuthenticationChallengeHandler)authenticatingHTTPProtocol:(JAHPAuthenticatingHTTPProtocol *)authenticatingHTTPProtocol
+                                        didReceiveAuthenticationChallenge:(NSURLAuthenticationChallenge *)challenge
+{
     self.authenticatingHTTPProtocol = authenticatingHTTPProtocol;
     
-    self.authAlertView = [[UIAlertView alloc] initWithTitle:@"JAHPDemo"
-                                                    message:@"Enter 'foo' for the username and 'bar' for the password"
-                                                   delegate:self
-                                          cancelButtonTitle:@"Cancel"
-                                          otherButtonTitles:
-                          @"OK",
-                          nil];
+    self.authAlertView =
+    [[UIAlertView alloc] initWithTitle: @"JAHPDemo"
+                               message: @"Enter 'foo' for the username and 'bar' for the password"
+                              delegate: self
+                     cancelButtonTitle: @"Cancel"
+                     otherButtonTitles: @"OK", nil];
+    
     self.authAlertView.alertViewStyle = UIAlertViewStyleLoginAndPasswordInput;
     [self.authAlertView show];
+    
+//    JAHPDidCancelAuthenticationChallengeHandler logCancelEvent =
```

Comment:
Why did you comment this out?

Reply:
That's what I've introduced before noticing the ```#define USE_PROTOCOL_FOR_CANCELLATION 0``` . Forgot to delete. Will fix that.